### ISSUE TITLE
[Feature request] Option to remove tracking etc

### ISSUE THREAD

**Issue by daggr8** [anonaddy/anonaddy#180]
Want to start with saying that I love AnonAddy. iI's  fantastic!

Duckduckgo just announced "Email Protection" for blocking trackers in mail. 
https://www.spreadprivacy.com/introducing-email-protection-beta/

Is this something that's on the roadmap? If even possible.

**Comment by willbrowningme** [comment]
Thanks for the suggestion, yes this should be possible.

I guess the hard part would be compiling a list of known tracking codes/domains used to track opens in emails.

Is anyone aware of any list already out there?

**Comment by daggr8** [comment]
I use these lists for my blocking on DNS level in my network. 
Maybe the Tracking & Telemetry Lists are usable?

**Comment by willbrowningme** [comment]
I think you forgot to include the link in your previous comment. Email pixel tracking domains would be the most appropriate to target, if I can start compiling a list of them.

**Comment by daggr8** [comment]
Of course i did. Sorry about that.
https://firebog.net/

More focused on email tracking there's these. Maybe there's more and better lists out there.

https://github.com/leavemealone-app/email-trackers
https://cerb.ai/guides/mail/privacy/block-email-tracking/
https://github.com/cfa999/email_trackers_block

**Comment by Coderdude112** [comment]
So the whole point with email pixel tracking is that they embed a image that is 1x1 pixels and then when your email client attempts to load this they can gather information on you (Opened time, aprox location, device, ect) correct?

So instead of using a block list to find and remove email trackers why not remove all images that are below a certain size? (Ex: All images with a height or width or 25 px are stripped out) And then use a list to determine what tracker it was similar to duckduckgo, that way even trackers that arn't on the block list will still be blocked.

This feature might even be improved with the ability to just remove all images regardless of their size.

**Comment by willbrowningme** [comment]
@Coderdude112 interesting suggestion which could work well, will look into it, thanks!

**Comment by ja7833** [comment]
I use https://email.faircode.eu/ which identifies tracking images (vs blocking like Duckduckgo beta).

I'm fairly certain this is done via https://disconnect.me/trackerprotection#categories-of-trackers available at https://github.com/disconnectme/disconnect-tracking-protection if you contact Marcel Bokhorst using https://contact.faircode.eu/?product=fairemailsupport he will probably respond within a day and be happy to provide some insight on how it works for https://play.google.com/store/apps/details?id=eu.faircode.email

A different solution to the issue than blocking but, perhaps one that could be implemented quickly?

**Comment by ashutoshsaboo** [comment]
+1 on this, this would be really an awesome feature to have natively especially for spam sites that like to profile their customers! Saw the last update here long back in 2021, @willbrowningme is this planned for anytime sooner?

**Comment by willbrowningme** [comment]
No update at the moment I'm afraid, I've got some other things I'm working on first then I can hopefully get around to adding this.

**Comment by joaonunovalente** [comment]
I guess it been a while @willbrowningme, but are thinking about coming back to this issue?

It would be a great feature.

**Comment by cscha** [comment]
+1

**Comment by cscha** [comment]
Any update? @willbrowningme

**Comment by benni347** [comment]
+1

**Comment by benni347** [comment]
I just found this page by GMass on how they got around [E-Mail tracking pixel blockers](https://www.gmass.co/blog/tracking-pixel-blockers/), tldr:
- they do set a predefined height/width so it has to be downloaded to get the size.
- Does not use query parameters
- Different length for the tracking pixel url
- Their server is hosted on aws with an aws domain.

**Comment by willbrowningme** [comment]
Yes I still plan to add this but just have a lot of other things to work on at the moment so cannot give an estimated date.

**Comment by luisnabais** [comment]
Hello! Just want to say that I also want this feature, is very important to me.

**Comment by emhl** [comment]
protonmail apparently uses https://github.com/DrKain/tidy-url to remove tracking links the filtering rules in this project could be a good starting off point

**Comment by Xynonners** [comment]
> So the whole point with email pixel tracking is that they embed a image that is 1x1 pixels and then when your email client attempts to load this they can gather information on you (Opened time, aprox location, device, ect) correct?
> 
> So instead of using a block list to find and remove email trackers why not remove all images that are below a certain size? (Ex: All images with a height or width or 25 px are stripped out) And then use a list to determine what tracker it was similar to duckduckgo, that way even trackers that arn't on the block list will still be blocked.
> 
> This feature might even be improved with the ability to just remove all images regardless of their size.

this is an ingenious way of doing it.

**Comment by benni347** [comment]
> So the whole point with email pixel tracking is that they embed a image that is 1x1 pixels and then when your email client attempts to load this they can gather information on you (Opened time, aprox location, device, ect) correct?
> 
> So instead of using a block list to find and remove email trackers why not remove all images that are below a certain size? (Ex: All images with a height or width or 25 px are stripped out) And then use a list to determine what tracker it was similar to duckduckgo, that way even trackers that arn't on the block list will still be blocked.
> 
> This feature might even be improved with the ability to just remove all images regardless of their size.

That is an intresting way to approach this problem, but I think it could lead to some data of the message being lost. For example a company could have multiple icons eg an envelop, phone whatever symbols to symbols some data eg "call us" with a phone icon next to it.

I hope it made sense what I was trying to say.

**Comment by Xynonners** [comment]
> > So the whole point with email pixel tracking is that they embed a image that is 1x1 pixels and then when your email client attempts to load this they can gather information on you (Opened time, aprox location, device, ect) correct?
> > So instead of using a block list to find and remove email trackers why not remove all images that are below a certain size? (Ex: All images with a height or width or 25 px are stripped out) And then use a list to determine what tracker it was similar to duckduckgo, that way even trackers that arn't on the block list will still be blocked.
> > This feature might even be improved with the ability to just remove all images regardless of their size.
> 
> That is an intresting way to approach this problem, but I think it could lead to some data of the message being lost. For example a company could have multiple icons eg an envelop, phone whatever symbols to symbols some data eg "call us" with a phone icon next to it.
> 
> I hope it made sense what I was trying to say.

You can replace it with a placeholder image, and also probably it'd be a good idea to allow it to be toggled per alias..

**Comment by emhl** [comment]
Toggling wether images are displayed, is something every decent email client already allows.

What the duckduckgo forwarding service does, that can't be done by an email client  is replacing the tracking links you can click on by following the redirect chain to get a non-personalized one

**Comment by shaunrileynet1** [comment]
This feature would be awesome.

**Comment by willbrowningme** [comment]
Currently in progress...

**Comment by willbrowningme** [comment]
This is now available for all those on Lite or Pro plans. You can also turn it off in your [account settings](https://app.addy.io/settings) under `Block Tracking Pixels`.

Also added a graph to the [dashboard](https://app.addy.io/) that shows the number of tracking pixels blocked over the last 7 days.

Any feedback is welcome!

**Comment by 2fsbb9op** [comment]
> Any feedback is welcome!

I like it!  Thanks for investing the time to implement a tracking pixel blocking feature for addy.

I only have handful of tracking pixels blocked ;)

But, as a thought **and only as a nice to know/nice to have feature in some future update** maybe the last 7 days graph could also show the top blocked domain or perhaps the top alias with blocked tracking pixels?
